### DISCUSSION TITLE
TODO

### DISCUSSION THREAD

**Discussion by zhongfly** [zhongfly/mpv-build#31] [General]
- [x] [job summaries](https://github.blog/2022-05-09-supercharging-github-actions-with-job-summaries/)
- [x] scoop or chocolatey

scoop example for x86_64-v3:
```
{
    "version": "2024-02-16-c0c32b2",
    "description": "Command line video player",
    "license": "GPL-2.0-or-later",
    "homepage": "https://mpv.io",
    "architecture": {
        "64bit": {
            "url": "https://github.com/zhongfly/mpv-winbuild/releases/download/2024-02-16-c0c32b2/mpv-x86_64-v3-20240216-git-c0c32b2.7z"
        }
    },
    "checkver": {
        "url": "https://api.github.com/repos/zhongfly/mpv-winbuild/releases/latest",
        "regex": "download/([\\w\\-]+)/mpv-x86_64-v3-(?<suffix>\\d+-git-\\w+)"
    },
    "autoupdate": {
        "architecture": {
            "64bit": {
                "url": "https://github.com/zhongfly/mpv-winbuild/releases/download/$version/mpv-x86_64-v3-$matchSuffix.7z"
            }
        }
    }
}
```

**Comment by Icey-Glitch** [comment]
@zhongfly Could you make a json that gets updated with the path of the latest release every build, this would make it way easier to make a manifest for scoop

**Reply by zhongfly** [reply]
Can you give me an example? I can follow the format of the example you gave

**Reply by zhongfly** [reply]
I have read scoop autoupdate wiki:https://github.com/ScoopInstaller/Scoop/wiki/App-Manifest-Autoupdate#special-cases
I think no need to make a json for every build.
```
"checkver": {
    "url": "https://api.github.com/repos/zhongfly/mpv-winbuild/releases/latest",
    "jsonpath": "$..browser_download_url",
    "regex": "download/(?<tag>[\\w\\-]+)/mpv-x86_64-(?<version>[\\d]+-git-\\w+)"
}
    "autoupdate": {
        "architecture": {
            "64bit": {
                "url": "https://github.com/zhongfly/mpv-winbuild/releases/download/$matchTag/mpv-x86_64-$matchVersion.7z#/dl.7z"
                "hash": {
                    "url": "https://github.com/zhongfly/mpv-winbuild/releases/download/$matchTag/sha256.txt"
                }
            }
    }
```
not tested
updated

**Reply by Icey-Glitch** [reply]
doesn't work (wasn't expecting that it would but just putting it out there). i am not sure how you intend it to work, the docs don't have anything comparable. could you explain

**Reply by zhongfly** [reply]
https://github.com/ScoopInstaller/Main/blob/master/bucket/git.json is a good example.
What I provided earlier is just a general reference for auto update and may have regex errors. From the documentation, it looks like using checkver and autoupdate will work perfectly fine,no need to make a json for you in every build.

**Reply by Icey-Glitch** [reply]
I was playing with it last night. I am not great at regex which is part of the problem. I just need to get the version with dashes now, the one referenced in the tag. I will send what I have currently if you are interested.

**Reply by zhongfly** [reply]
I have updated regex,you can try it again.What you want is `$matchTag`

**Reply by Icey-Glitch** [reply]
Hello,
thank you for your help. I have successfully created a manifest here is the working auto update / checkver.
```json
"checkver": {
    "github": "https://github.com/zhongfly/mpv-winbuild",
    "jsonpath": "$.assets[11]..browser_download_url",
    "regex": "download\\/(?<dash>(?:\\d{4}-(?:0[1-9]|1[0-2])-(?:0[1-9]|[12]\\d|3[01])))-(?<commit>\\w+)\\/mpv-x86_64-v3-(?<version>\\d+)-git-\\k<commit>\\.7z"
},
"autoupdate": {
    "architecture": {
        "64bit": {
            "url": "https://github.com/zhongfly/mpv-winbuild/releases/download/$matchDash-$matchCommit/mpv-x86_64-v3-$version-git-$matchCommit.7z",
            "hash": {
                "url": "https://github.com/zhongfly/mpv-winbuild/releases/download/$matchDash-$matchCommit/sha256.txt",
                "regex": "^(?<sha256>[a-f0-9]{64})\\s+\\S*mpv-x86_64-v3-(?<version>\\d+)-git-(?<commit>[a-f0-9]+)\\.7z\\s*\\d*"
            }
        }
    }
}
```

**Reply by zhongfly** [reply]
I found the jsonpath is `$.assets[11]`,is the x86_64-v3 mpv always in assets[11]?
A fixed number is not good.

**Reply by Icey-Glitch** [reply]
I do not know another way. unless you change something it should be fine
scoop doesn't allow more than one string i think and i cant figure out json path filtering
`could not find '$..browser_download_url' in https://api.github.com/repos/zhongfly/mpv-winbuild/releases/latest`

**Reply by zhongfly** [reply]
https://github.com/search?q=repo%3AScoopInstaller%2FMain%20api.github.com&type=code
Scoop bucket always use `api.github.com/repositories/` instead of `api.github.com/repos/`
Maybe we can try this one `https://api.github.com/repositories/412382551/releases/latest`(same json as `https://api.github.com/repos/zhongfly/mpv-winbuild/releases/latest`)
```
    "url": "https://api.github.com/repositories/412382551/releases/latest",
    "jsonpath": "$..browser_download_url",
```

**Reply by zhongfly** [reply]
https://github.com/search?q=repo%3AScoopInstaller%2FExtras%20api.github.com&type=code
Or `"jsonpath": "$.assets..browser_download_url"`
Or we can try to remove `jsonpath` and only use `regex`

**Reply by zhongfly** [reply]
And I suggest using tag name as version code instead of only date,because there may be multiple releases a day.
```
    "checkver": {
        "url": "https://api.github.com/repos/zhongfly/mpv-winbuild/releases/latest",
        "regex": "download/([\\w\\-]+)/mpv-x86_64-v3-(?<suffix>\\d+-git-\\w+)"
    },
    "autoupdate": {
        "architecture": {
            "64bit": {
                "url": "https://github.com/zhongfly/mpv-winbuild/releases/download/$version/mpv-x86_64-v3-$matchSuffix.7z"
            }
        }
    }
```
works fine for me with scoop 0.3.1

**Reply by Icey-Glitch** [reply]
I can change it to be hash if that would be better but it should not break still because of how the hash works it will still find the named captures to be diff as-well

**Reply by Icey-Glitch** [reply]
this is what i decided on
```json
"checkver": {
    "github": "https://github.com/zhongfly/mpv-winbuild",
    "regex": "download\\/(?<version>(?<dash>(?:\\d{4}-(?:0[1-9]|1[0-2])-(?:0[1-9]|[12]\\d|3[01])))-(?<commit>\\w+))\\/mpv-x86_64-v3-(?<cleanversion>\\d+)-git-\\k<commit>\\.7z"
},
"autoupdate": {
    "architecture": {
        "64bit": {
            "url": "https://github.com/zhongfly/mpv-winbuild/releases/download/$matchDash-$matchCommit/mpv-x86_64-v3-$matchCleanversion-git-$matchCommit.7z",
            "hash": {
                "url": "https://github.com/zhongfly/mpv-winbuild/releases/download/$matchDash-$matchCommit/sha256.txt",
                "regex": "^(?<sha256>[a-f0-9]{64})\\s+\\S*mpv-x86_64-v3-(?<version>\\d+)-git-(?<commit>[a-f0-9]+)\\.7z\\s*\\d*"
            }
        }
    }
}
```
I think it might be a little over done but i think it will be easier to work on
